Mac build

bam241 opened the pull request:
this allow to build.on Mac OS 

it also an autogenerated version.hpp file for the version numbers.

bam241 commented:
when compiling on Mac OS with non-standalone mode I got:
```
Undefined symbols for architecture x86_64:
"_Gopt", referenced from: [...]
```
which is solved by defining `struct program_option_struct Gopt;` which is defined original in `option.hpp` as extern.

bam241 commented:
@gonuke @pshriwise @ljacobson64 this should be ready to be reviewed.

gonuke commented on version.hpp.in:
```diff
@@ -0,0 +1,10 @@
+#define MCNP2CAD_VERSION_MINOR @MCNP2CAD_VERSION_MINOR@
+#define MCNP2CAD_VERSION_REV @MCNP2CAD_VERSION_PATCH@
+
+#define STANDALONE_MCNP2CAD '@STANDALONE_MCNP2CAD@'
```
Is there a reason for quotes here?

bam241 replied:
I don't quite remember.... I'll check

bam241 replied:
I am not sure...
I'll try without.

gonuke commented on CMakeLists.txt:
```diff
@@ -90,8 +85,10 @@ set(MCNP2CAD_PUB_HEADERS dataref.hpp
                          mcnp2cad.hpp
                          MCNPInput.hpp
                          options.hpp
-                         version.hpp
```
don't we still want this to be installed even though it is autogenerated?

bam241 replied:
we do at line 91

bam241 replied:
I think ?!

gonuke replied:
I think line 91 makes it available for the build, but doesn't install it for use by downstream applications, although I'm not 100% sure.

bam241 replied:
yes you are probably right...

gonuke reviewed (changes requested):
Just a couple of little questions/changes and then happy to merge

ljacobson64 commented on CMakeLists.txt:
```diff
@@ -18,15 +22,6 @@ set(CMAKE_CXX_STANDARD 11)
-if (CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
-    if (NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
-      set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
```
Perhaps I am out of the loop, but how come these lines were removed? I remember them being necessary when I was building on Linux.

bam241 replied:
it is still needed for Trelis 17.1.
And we now add it manually on the reps we build (MOAB, and DAGMC).

but as this need to be removed for Cubit 2020.2, we decided that we remove them and one specify them manually in the make process.
(@pshriwise I think we need to update the doc)

bam241 commented:
@gonuke I think I addressed all your comments.

(@pshriwise @gonuke this is the next PR to get reviewed/merged in the Trellis-plugin GitHub Action chain.)

gonuke commented:
In the absence of CI, I'll merge this for testing downstream
